modules/apps/callbacks/ica_test.go

```diff
@@ -0,0 +1,197 @@
+			fmt.Sprintf(`{"dest_callback": {"address": "%s"}}`, callbackAddr),
+			"none",
+			true,
+		},
+		{
+			"success: source callback",
+			fmt.Sprintf(`{"src_callback": {"address": "%s"}}`, callbackAddr),
+			types.CallbackTypeAcknowledgementPacket,
+			true,
+		},
+		{
+			"success: source callback with other json fields",
+			fmt.Sprintf(`{"src_callback": {"address": "%s"}, "something_else": {}}`, callbackAddr),
+			types.CallbackTypeAcknowledgementPacket,
+			true,
+		},
+		{
+			"success: source callback with malformed json",
+			fmt.Sprintf(`{"src_callback": {"address": "%s"}, malformed}`, callbackAddr),
+			"none",
+			true,
+		},
+		{
+			"success: source callback with missing address",
+			`{"src_callback": {"address": ""}}`,
+			"none",
+			true,
+		},
+		{
+			"failure: source callback with low gas (panic)",
+			fmt.Sprintf(`{"src_callback": {"address": "%s", "gas_limit": "350000"}}`, callbackAddr),
+			types.CallbackTypeSendPacket,
+			false,
+		},
+	}
+
+	for _, tc := range testCases {
+		s.Run(tc.name, func() {
+			icaAddr := s.SetupICATest()
+
+			s.ExecuteICATx(icaAddr, tc.icaMemo, 1)
+			s.AssertHasExecutedExpectedCallback(tc.expCallback, tc.expSuccess)
+		})
+	}
+}
+
+func (s *CallbacksTestSuite) TestICATimeoutCallbacks() {
+	// ICA channels are closed after a timeout packet is executed
+	testCases := []struct {
+		name        string
+		icaMemo     string
+		expCallback types.CallbackType
+		expSuccess  bool
+	}{
+		{
+			"success: send ica tx timeout with no memo",
+			"",
+			"none",
+			true,
+		},
+		{
+			"success: dest callback",
+			fmt.Sprintf(`{"dest_callback": {"address": "%s"}}`, callbackAddr),
+			"none",
+			true,
+		},
+		{
+			"success: source callback",
+			fmt.Sprintf(`{"src_callback": {"address": "%s"}}`, callbackAddr),
+			types.CallbackTypeTimeoutPacket,
+			true,
+		},
+		{
+			"failure: source callback with low gas (panic)",
+			fmt.Sprintf(`{"src_callback": {"address": "%s", "gas_limit": "350000"}}`, callbackAddr),
+			types.CallbackTypeSendPacket,
+			false,
+		},
+	}
+
+	for _, tc := range testCases {
+		s.Run(tc.name, func() {
+			icaAddr := s.SetupICATest()
+
+			s.ExecuteICATimeout(icaAddr, tc.icaMemo, 1)
+			s.AssertHasExecutedExpectedCallback(tc.expCallback, tc.expSuccess)
+		})
+	}
+}
+
+// ExecuteICATx executes a stakingtypes.MsgDelegate on chainB by sending a packet containing the msg to chainB
+func (s *CallbacksTestSuite) ExecuteICATx(icaAddress, memo string, seq uint64) {
+	timeoutTimestamp := uint64(s.chainA.GetContext().BlockTime().Add(time.Minute).UnixNano())
+	icaOwner := s.chainA.SenderAccount.GetAddress().String()
+	connectionID := s.path.EndpointA.ConnectionID
+	// build the interchain accounts packet data
+	packetData := s.buildICAMsgDelegatePacketData(icaAddress, memo)
+	msg := icacontrollertypes.NewMsgSendTx(icaOwner, connectionID, timeoutTimestamp, packetData)
+
+	res, err := s.chainA.SendMsgs(msg)
+	if err != nil {
+		return // we return if send packet is rejected
+	}
```

Comment:
is there testcases which require this to no-op here? Otherwise we can just require no error, same for the func below - `ExecuteICATimeout`

Reply:
Yes, there are test cases that require this. Mainly for when we wanna test that SendPacket is really blocking on errors